src/app/checkout/navigateToOrderConfirmation.tsx

```diff
@@ -1,7 +1,18 @@
 import { noop } from 'lodash';
 
+interface RouteKeyMap<T = string> {
+    [key: string]: T;
+}
+
+const orderConfirmationRoutesMap: RouteKeyMap = {
+    '/embedded-checkout': '/embedded-checkout/order-confirmation',
+};
+
+const defaultOrderConfirmationRoute = '/checkout/order-confirmation';
+
 export default function navigateToOrderConfirmation(location = window.location): Promise<never> {
-    const url = `${location.pathname.replace(/\/$/, '')}/order-confirmation`;
+    const orderConfirmationRoute = Object.keys(orderConfirmationRoutesMap).filter(route => location.pathname.includes(route));
```

Comment:
You can use `find` here?
```suggestion
    const orderConfirmationRoute = Object.keys(orderConfirmationRoutesMap).find(route => location.pathname.includes(route));
```

That way you don't have to `orderConfirmationRoutesMap[orderConfirmationRoute[0]]` in the next line. You can just do `orderConfirmationRoutesMap[orderConfirmationRoute]` in the next line.

Reply:
Thanks for reviewing. Currently, we only have two routes for rendering checkout-js (`/checkout` and `/embedded-checkout`). Maybe we could introduce mapping when we have an additional route.